dbt/macros/get_s3_dependency_dir.sql

```diff
@@ -0,0 +1,48 @@
+-- Get the S3 location that is used to store Python model dependencies
+-- for the current target
```

Comment:
Ideally we would just add a new attribute `s3_dependency_dir` to our profiles defined in `profiles.yml`, and then we could have full control over the location where dependencies get stored for each target. That doesn't appear to be possible without extending the `dbt-athena` plugin itself to support a new profile variable, so this macro represents a workaround that allows us to compute a dependency dir based on the `s3_data_dir` variables set in our profiles. 

I think we'll probably need a Python version of this macro as well, but I'm leaving that for myself to figure out later when I tackle https://github.com/ccao-data/data-architecture/issues/439.